src/cata_tiles.cpp

```diff
@@ -2314,7 +2314,12 @@ bool cata_tiles::draw_entity( const Creature &critter, const tripoint &p, lit_le
             std::string chosen_id = ent_name.str();
             if( m->has_effect( effect_ridden ) ) {
                 int pl_under_height = 6;
-                draw_entity_with_overlays( g->u, p, ll, pl_under_height );
+                if( m->mounted_player ) {
+                    player *pl = m->mounted_player;
+                    if( pl ) {
```

Comment:
This check is pointless. `mounted_player` is a `player*`, so if it was non-null in the check above, the copy of of (`pl`) will be non-null as well.